scripts/collect_workloads_agent.py

```
@@ -0,0 +1,618 @@
+# ──────────────────────────────────────────────────────────────────────────────
+# Tool implementations
+# ──────────────────────────────────────────────────────────────────────────────
+
+REPO_ROOT = Path(__file__).parent.parent
+CONDA_ENV = os.environ.get("CONDA_ENV", "flashinfer_bench")
+
+
+def _run(cmd: str, timeout: int = 60) -> dict:
+    """Run a shell command, return {stdout, stderr, returncode}."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)